opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -476,6 +480,24 @@ namespace Opm {
         }
     }
 
+    void Schedule::handleWSOLVENT(DeckKeywordConstPtr keyword, size_t currentStep) {
+        for (size_t recordNr = 0; recordNr < keyword->size(); recordNr++) {
+            DeckRecordConstPtr record = keyword->getRecord(recordNr);
+            const std::string& wellNamePattern = record->getItem("WELL")->getTrimmedString(0);
+            std::vector<WellPtr> wells = getWells(wellNamePattern);
+
+            for (auto wellIter=wells.begin(); wellIter != wells.end(); ++wellIter) {
+                WellPtr well = *wellIter;
+
+                WellSolventProperties properties(well->getSolventPropertiesCopy(currentStep));
+
+                properties.m_solventConcentration = record->getItem("SOLVENT_FRACTION")->getSIDouble(0);
```

Comment:
actually I think SI double is a bit better: who knows how Eclipse likes to specify fractions... (I would only slightly surprised if they sometimes use "parts of 42" or something.)